src/handlers/AdminAggregatorMessageHandler.ts

```diff
@@ -0,0 +1,125 @@
+import EventId from '../utils/eventId';
+import InstanceAuthority from '../authorities/InstanceAuthority';
+import {Ps2alertsEventState} from '../ps2alerts-constants/ps2alertsEventState';
+import {getLogger} from '../logger';
+import {jsonLogOutput} from '../utils/json';
+import AdminAggregatorInstanceStartMessage from '../data/AdminAggregator/AdminAggregatorInstanceStartMessage';
+import AdminAggregatorInstanceEndMessage from '../data/AdminAggregator/AdminAggregatorInstanceEndMessage';
+import {Bracket} from '../ps2alerts-constants/bracket';
+import {ChannelActionsInterface, QueueMessageHandlerInterface} from '../interfaces/QueueMessageHandlerInterface';
+import ExceptionHandler from './system/ExceptionHandler';
+
+@injectable()
+export default class AdminAggregatorMessageHandler implements QueueMessageHandlerInterface<AdminQueueMessage> {
+    private static readonly logger = getLogger('AdminAggregatorMessageHandler');
+
+    constructor(private readonly instanceAuthority: InstanceAuthority) {}
+
+    public async handle(event: AdminQueueMessage, actions: ChannelActionsInterface): Promise<void> {
+        switch (event.type) {
+            case 'instanceStart':
+                return await this.startInstance(event, actions);
+            case 'instanceEnd':
+                return await this.endInstance(event, actions);
+            case 'endAll':
+                return await this.endAllInstances(actions);
+            case 'activeInstances':
+                return this.activeInstances(actions);
+        }
+
+        throw new ApplicationException(`Unknown AdminAggregator message received: ${jsonLogOutput(event)}`, 'AdminAggregatorMessageHandler');
+    }
+
+    // Collect the required information from the message in order to generate an PS2AlertsMetagameInstance, then trigger it.
+    // This purposefully bypasses most of the restrictions of starting an instance in case of events or debugging.
+    private async startInstance(message: AdminQueueMessage, actions: ChannelActionsInterface): Promise<void> {
+        const adminAggregatorInstanceStart = new AdminAggregatorInstanceStartMessage(message.body);
+
+        const censusEventId = EventId.zoneFactionMeltdownToEventId(
+            adminAggregatorInstanceStart.zone,
+            adminAggregatorInstanceStart.faction,
+            adminAggregatorInstanceStart.meltdown,
+        );
+
+        const metagameDetails = metagameEventTypeDetailsMap.get(censusEventId);
+
+        if (!metagameDetails) {
+            // eslint-disable-next-line @typescript-eslint/restrict-template-expressions
+            AdminAggregatorMessageHandler.logger.error(`Zone: ${adminAggregatorInstanceStart.zone} - Faction: ${adminAggregatorInstanceStart.faction} - Meltdown: ${adminAggregatorInstanceStart.meltdown}`);
+
+            throw new ApplicationException(`Unknown metagame event id ${censusEventId}`, 'AdminAggregatorMessageHandler');
```

Comment:
So basically any uncaught exceptions within the handlers are caught at the queue level. In the case for the AdminMessageHandler, they're caught in the AdminQueue.

https://github.com/ps2alerts/aggregator/blob/efb5b99a31f6cf24d333c7fd37b671efddfd772a/src/services/rabbitmq/queues/AdminQueue.ts#L65

Reply:
Okay, but what in the case of a resolve without a ack or nack?

Reply:
I see what you're trying to say - there should be a default ack at the queue level. However, I foresee an issue where messages will be acked, and then acked again.

E.g. if we put `channel.ack()` in say here in `InstanceEventQueue`:

```ts
await this.timingMiddlewareHandler.handle(
                            this.createPs2Event(message),
                            {
                                ack: () => this.handleMessageConfirm(message, 'ack'),
                                retry: () => this.handleMessageConfirm(message, 'retry'),
                                // delay: (ms: number) => this.handleMessageDelay(message, ms, this.pattern),
                                delay: () => {
                                    return true;
                                }, // TODO: IMPLEMENT
                            },
                            this.handler,
                        );
                        channel.ack(message) // <<< here
```
Then that would mean we cannot ack within the handlers. That might be a right pattern to approach perhaps, and in the case of `ZoneMessageHandler` we are doing exactly that, we're awaiting then acking here: 

https://github.com/ps2alerts/aggregator/blob/efb5b99a31f6cf24d333c7fd37b671efddfd772a/src/handlers/ps2census/ZoneMessageHandler.ts#L46

I chose not to force the acking because the implementation of the acking may depend on the queue.

What I am open to is perhaps wrapping the await <handler> call to the abstract class, and within that have something like

```ts
protected async sendToHandlers(handler) {
  // send to handlers with actions inserted
    
  const promise === handler.handle etc etc
  await Promise.race(promise)
  
  try {
    return actions.ack()
  }
  catch (e) {
    // Handle "already acked" issue here
  }
}
```

That may work

Reply:
Maybe just like the timing you can use a proxy and a variable that allows you to check if any of the actions is called, might eliminate any side-effects completely.